### PR TITLE
[STORM-1143] - Validate topology Configs during topology submission

### PR DESCRIPTION
**Pull request by jerrypeng**
adding config validation during the topology submission process so that incorrect configs can be detected early

### CODE REVIEW

**Comment by jerrypeng** [comment]
non related failure in travis for jdk7. Tests for jdk 8 passed

File: storm-core/src/jvm/backtype/storm/validation/ConfigValidation.java [hunk 1]
```diff
@@ -504,42 +509,45 @@ public static void validateField(String fieldName, Map conf, Class configClass)
...
+                    //If not call default constructor
+                    else {
+                        o = (((Class<Validator>) clazz).newInstance());
+                    }
+                    o.validateField(field.getName(), conf.get(key));
                 }
-                o.validateField(field.getName(), conf.get(key));
             }
+        }  catch (NoSuchMethodException | IllegalAccessException | InstantiationException | InvocationTargetException e) {
+            throw new IllegalStateException("ConfigValidation failed due to: " + e.getMessage() + "\n" + e.getStackTrace().toString());
```
Comment (d2r) [inline storm-core/src/jvm/backtype/storm/validation/ConfigValidation.java]: Would this be a operation done in an illegal state?  Maybe we could just wrap `e` in a RuntimeException?

File: storm-core/src/jvm/backtype/storm/validation/ConfigValidation.java [hunk 2]
```diff
@@ -558,9 +566,14 @@ public static void validateFields(Map conf) throws IllegalAccessException, Insta
...
      */
-    public static void validateFields(Map conf, Class configClass) throws IllegalAccessException, InstantiationException, NoSuchFieldException, NoSuchMethodException, InvocationTargetException {
+    public static void validateFields(Map conf, Class configClass) {
         for (Field field : configClass.getFields()) {
-            Object keyObj = field.get(null);
+            Object keyObj = null;
+            try {
+                keyObj = field.get(null);
+            } catch (IllegalAccessException e) {
+                throw new IllegalStateException("ConfigValidation failed due to: " + e.getMessage() + "\n" + e.getStackTrace().toString());
```
Comment (d2r) [inline storm-core/src/jvm/backtype/storm/validation/ConfigValidation.java]: Same here, if we just don't want checked exceptions, we can wrap `e` in a RTE.

**Comment by jerrypeng** [comment]
@d2r  thanks for your review, I just pushed a commit that I think addressed your comments

File: storm-core/src/jvm/backtype/storm/StormSubmitter.java [hunk 3]
```diff
@@ -18,6 +18,7 @@
 package backtype.storm;
 
 import java.io.File;
+import java.lang.reflect.InvocationTargetException;
```
Comment (d2r) [inline storm-core/src/jvm/backtype/storm/StormSubmitter.java]: remove?

Comment (jerrypeng) [inline storm-core/src/jvm/backtype/storm/StormSubmitter.java]: will remove

**Comment by d2r** [comment]
One minor nit, then I am +1 pending relevant tests pass.

**Comment by jerrypeng** [comment]
@d2r thanks again! Just push a commit to fix the one nit

**Comment by jerrypeng** [comment]
Tests passed for java 8. Unrelated test failure for java 7 build

**Comment by knusbaum** [comment]
It doesn't look like `ConfigValidation` has changed functionally at all. Is that correct?

**Comment by jerrypeng** [comment]
@knusbaum the functionality of ConfigValidation has not changed, but this PR adds support for validating configs submitted with a topology.  This allows storm to do early detection on whether the configs submitted with a topology are valid

**Comment by revans2** [comment]
+1 looks good to me.

**Comment by knusbaum** [comment]
I guess I'm wondering why all those changes are in `ConfigValidation`. 
I'm +1, but I'd rather have changes like that in a separate pull request for a separate jira, like 'remove checked exceptions' or something.

**Pull request merged**